### DISCUSSION TITLE
Migrating from self hosted to docker?

### DISCUSSION THREAD

**Discussion by Plutoaurus** [firefly-iii/firefly-iii#8289] [Firefly III]
After Firefly upgraded php version I've decided it's probably easier to replace my self hosted setup to docker so future updates are easier.

Is there any guidance on migrating all transactions to a new server?

Thanks!!!

**Comment by JC5** [comment]
There's no easy guide yet, but basically, use mysqldump to export your database from your self-hosted database. Then, set up Firefly III (the same version!), make sure you get the registration screen and then import the same mysqldump again.

You can do that by using GUI tools like Querious on the Mac or something else on Windows, or by using the command line.

**Reply by Plutoaurus** [reply]
OK ta,  will try it out.

Have a good Christmas!

**Comment by papa-fuchs** [comment]
i did it today via mysqldump it works fine at the end.
just a few commands that helped me alot:

```
# search container ID (docker)
docker ps | grep -E 'firefly[-_](iii|)[_-]?db' | cut -d ' ' -f 1

# backup DB (self hosted)
bash -c '/usr/bin/mysqldump -u DBUSER --password="DBPASSWORD" "DBNAME"' > "/mnt/backup/firefly_db.sql"

#restore DB (docker)
cat "/mnt/backup/firefly_db.sql" | docker exec -i CONTAINERID bash -c '/usr/bin/mariadb -u DBUSER --password="DBPASSWORD" "DBNAME"'
```

**Reply by webmasterMeyers** [reply]
Thanks for sharing the commands back here! I will probably be migrating a few times.
